public/app/controllers/resources_controller.rb

```diff
@@ -205,62 +201,62 @@ def waypoints
+	rescue RecordNotFound
+		render json: {}, status: 404
+	end
 
   def tree_node_from_root
-    @root_uri = "/repositories/#{params[:rid]}/resources/#{params[:id]}"
-
-    render :json => archivesspace.get_raw_record(@root_uri + '/tree/node_from_root_' + params[:node_ids].first)
+		@root_uri = "/repositories/#{params[:rid]}/resources/#{params[:id]}"
+		render :json => archivesspace.get_raw_record(@root_uri + '/tree/node_from_root_' + params[:node_ids].first)
+	rescue RecordNotFound
+		render json: {}, status: 404
   end
 
   def inventory
-    uri = "/repositories/#{params[:rid]}/resources/#{params[:id]}"
-
-    tree_root = archivesspace.get_raw_record(uri + '/tree/root') rescue nil
-    @has_children = tree_root && tree_root['child_count'] > 0
-
-    begin
-      # stuff for the collection bits
-      @criteria = {}
-      @criteria['resolve[]']  = ['repository:id', 'resource:id@compact_resource', 'top_container_uri_u_sstr:id', 'related_accession_uris:id']
-      @result =  archivesspace.get_record(uri, @criteria)
-      @repo_info = @result.repository_information
-      @page_title = "#{I18n.t('resource._singular')}: #{strip_mixed_content(@result.display_string)}"
-      @context = [{:uri => @repo_info['top']['uri'], :crumb => @repo_info['top']['name']}, {:uri => nil, :crumb => process_mixed_content(@result.display_string)}]
-      fill_request_info
-
-      # top container stuff ... sets @records
-      fetch_containers(uri, "#{uri}/inventory", params)
-
-      if !@results.blank?
-        params[:q] = '*'
-        @pager =  Pager.new(@base_search, @results['this_page'], @results['last_page'])
-      else
-        @pager = nil
-      end
-
-    rescue RecordNotFound
-      @type = I18n.t('resource._singular')
-      @page_title = I18n.t('errors.error_404', :type => @type)
-      @uri = uri
-      @back_url = request.referer || ''
-      render  'shared/not_found', :status => 404
-    end
+		uri = "/repositories/#{params[:rid]}/resources/#{params[:id]}"
+		tree_root = archivesspace.get_raw_record(uri + '/tree/root') rescue nil
+		@has_children = tree_root && tree_root['child_count'] > 0
+		# stuff for the collection bits
+		@criteria = {}
+		@criteria['resolve[]']  = ['repository:id', 'resource:id@compact_resource', 'top_container_uri_u_sstr:id', 'related_accession_uris:id']
+		@result =  archivesspace.get_record(uri, @criteria)
+		@repo_info = @result.repository_information
+		@page_title = "#{I18n.t('resource._singular')}: #{strip_mixed_content(@result.display_string)}"
+		@context = [{:uri => @repo_info['top']['uri'], :crumb => @repo_info['top']['name']}, {:uri => nil, :crumb => process_mixed_content(@result.display_string)}]
+		fill_request_info
+
+		# top container stuff ... sets @records
+		fetch_containers(uri, "#{uri}/inventory", params)
+
+		if !@results.blank?
+			params[:q] = '*'
+			@pager =  Pager.new(@base_search, @results['this_page'], @results['last_page'])
+		else
+			@pager = nil
+		end
+
+	rescue RecordNotFound
```

Comment:
Personally not a fan of the implicit rescue for a method (vs explicit `begin` - `rescue` block). But if y'all happy with it I guess I can be too 😄

Reply:
I would prefer an explicit begin - rescue block as well